clients/src/main/java/org/apache/kafka/clients/MetadataCache.java

```diff
@@ -69,13 +73,20 @@ private MetadataCache(String clusterId,
                           Set<String> invalidTopics,
                           Set<String> internalTopics,
                           Node controller,
+                          Map<String, Uuid> topicIds,
                           Cluster clusterInstance) {
         this.clusterId = clusterId;
         this.nodes = nodes;
         this.unauthorizedTopics = unauthorizedTopics;
         this.invalidTopics = invalidTopics;
         this.internalTopics = internalTopics;
         this.controller = controller;
+        this.topicIds = topicIds;
+
+        this.topicNames = new HashMap<>(topicIds.size());
```

Comment:
As far as I can tell, there are no uses of this collection (`Metadata.topicName` is not referenced). Can we get rid of it?

Reply:
sure